### ISSUE TITLE
Value for `sonar.branch.target` in `release` and `master` branches?

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#2095]
**Bug Description**

There is a comment in [`sonar-project.properties`](https://github.com/bitshares/bitshares-core/blob/851a97e598a323f06995a420f511e1f355cd1cd6/sonar-project.properties#L16-L17):
```
# should be changed in hardfork branch and removed in release branches
sonar.branch.target=develop
```

Although I don't know what the parameter is and why the comment says so, it seems incorrect that we have that value in `master` branch. I think
<del>1. when creating new release we need to update the value in `master` branch, and</del> (update: 2 will obsolete 1)
2. ideally we should have a build helper script to update the value when building the project, so that we don't need to update it every time when creating a new branch, nor need to fix conflicts caused by it when merging branches.

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by this bug. Please tick at least one box.
- [ ] API (the application programming interface)
- [x] Build (the build process or something prior to compiled code)
- [ ] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [ ] Performance (system or user efficiency, etc.)
- [ ] Protocol (the blockchain logic, consensus, validation, etc.)
- [ ] Security (the security of system or user data, etc.)
- [ ] UX (the User Experience)
- [ ] Other (please add below)

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Bug Report
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

**Comment by abitmore** [comment]
Background: we introduced a change in #1750 that requires some actions when merging branches, but haven't documented it in [Git Flow](https://github.com/bitshares/bitshares-core/wiki/Git-Flow), so I guess nobody remembered it.

**Comment by abitmore** [comment]
There is a related [build helper script](https://github.com/bitshares/bitshares-core/blob/master/programs/build_helpers/set_sonar_branch) in the project.

**Comment by abitmore** [comment]
Fixed by #2102.
